streams/shared/src/main/scala/zio/stream/ZPipeline.scala

```diff
@@ -1788,6 +1788,41 @@ object ZPipeline extends ZPipelinePlatformSpecificConstructors {
       ZChannel.identity[Nothing, Chunk[In], Any].concatMap(_.map(f).map(_.channel).fold(ZChannel.unit)(_ *> _))
     )
 
+  /**
+   * Creates a pipeline that maps chunks of elements with the specified
+   * function.
+   *
+   * Will stop on the first Left found
+   */
+  def mapEitherChunked[Env, Err, In, Out](
+    f: In => Either[Err, Out]
+  )(implicit trace: Trace): ZPipeline[Env, Err, In, Out] = {
+    lazy val reader: ZChannel[Env, Err, Chunk[In], Any, Err, Chunk[Out], Any] =
+      ZChannel.readWithCause(
+        chunk => {
```

Comment:
Since destroying chunking is not uncommon, I'm wondering whether we should shortcut `chunk.size == 1`

Reply:
It's a good idea! Thanks :)

Done